src/prefect/blocks/core.py

```diff
@@ -272,6 +273,19 @@ class Block(BaseModel, ABC):
     )
 
     def __init__(self, *args, **kwargs):
+        block_document_id = kwargs.pop("$ref", None)
+        if block_document_id:
+            block_document, block_document_name = self._get_block_document_by_id(
+                block_document_id, _sync=True
+            )
```

Comment:
I'm wary of putting a blocking API call in `__init__`. Could you add a new method to load a block from an ID? That would require an update to `Flow.validate_parameters` to use that new method, but I think that'd make this new functionality more explicit.

Reply:
Good call, agreed on this -- maybe it could be a classmethod `load_from_ref` that the `validate_parameters` method calls explicitly for Block typed parameters.

Reply:
Refactored to new method `load_from_ref` which is used for preprocessing parameters in `validate_parameters`. To deal with duplicate code i dared to create `_load_from_block_document` method to be used both by `load` and by `load_from_ref`.